EIPS/eip-3026.md

```diff
@@ -0,0 +1,287 @@
+
+## Rationale
+Motivation section covers a total motivation to have operations over BW6-761 curve available. We also extend a rationale for more specific fine points.
+
+#### Multiexponentiation as a separate call
+
+Explicit separate multiexponentiation operation that allows one to save execution time (so gas) by both the algorithm used (namely Peppinger algorithm) and (usually forgotten) by the fact that `CALL` operation in Ethereum is expensive (at the time of writing), so one would have to pay non-negigible overhead if e.g. for multiexponentiation of `100` points would have to call the multipication precompile `100` times and addition for `99` times (roughly `138600` would be saved).
+
+#### Explicit subgroup checks
+G2 subgroup check has the same cost as G1 subgroup check. Endomorphisms can be leverages to optimize this operation.
+
+## Backwards Compatibility
+There are no backward compatibility questions.
+
+## Test Cases
+
+Due to the large test parameters space we first provide properties that various operations must satisfy. We use additive notation for point operations, capital letters (`P`, `Q`) for points, small letters (`a`, `b`) for scalars. Generator for G1 is labeled as `G`, generator for G2 is labeled as `H`, otherwise we assume random point on a curve in a correct subgroup. `0` means either scalar zero or point of infinity. `1` means either scalar one or multiplicative identity. `group_order` is a main subgroup order. `e(P, Q)` means pairing operation where `P` is in G1, `Q` is in G2.
+
+Requeired properties for basic ops (add/multiply):
+
+- Commutativity: `P + Q = Q + P`
+- Additive negation: `P + (-P) = 0`
+- Doubling `P + P = 2*P`
+- Subgroup check: `group_order * P = 0`
+- Trivial multiplication check: `1 * P = P`
+- Multiplication by zero: `0 * P = 0`
+- Multiplication by the unnormalized scalar `(scalar + group_order) * P = scalar * P`
+
+Required properties for pairing operation:
+- Degeneracy `e(P, 0*Q) = e(0*P, Q) = 1`
+- Bilinearity `e(a*P, b*Q) = e(a*b*P, Q) = e(P, a*b*Q)` (internal test, not visible through ABI)
+
+Test vector for all operations are expanded in this [gist](https://gist.github.com/shamatar/506ab3193a7932fe9302a2f3a31a23e8) until it's final.
+
+## Implementation
+There is a various choice of existing implementations:
+**Libraries:**
+- Rust implementation (EY/Zexe): https://github.com/yelhousni/zexe/tree/youssef/BW6-761-Fq-ABLR-2ML-M
+- C++ implementation (EY/libff): https://github.com/EYBlockchain/zk-swap-libff
+- Golang implementation (Consensys/gurvy): https://github.com/ConsenSys/gurvy
+
+**Stand-alone implementation:**
+- Golang implementation with Intel assembly (Onur Kilic): https://github.com/kilic/bw6
+
+**Precompiles:**
+- OpenEthereum (EY/Parity): https://github.com/EYBlockchain/solidity-elliptic-curves
+
+**Sripts:**
+- SageMath and Magma scripts: https://gitlab.inria.fr/zk-curves/bw6-761/
```

Comment:
This section is no longer required.  Consider just including a reference implementation written in a language that is easy for a human to read and is self contained.  We try to avoid external links whenever possible since they tend to 404 with time, so an inline or `../assets/eip-3026/reference_implementation.whatever` would be preferred to an external implementation.

Reply:
This sounds good, but I would like to keep the implementations links for now until we agree upon a reference implementation. There are few implementations with different performances.